src/cpp/tests/cpp/tests/fixtures/RequiresPrivilegeTestFixture.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * RequiresPrivilegeTestFixture.hpp
+ *
+ * Copyright (C) 2023 by Posit Software, PBC
+ *
+ * Unless you have received this program directly from Posit Software pursuant
+ * to the terms of a commercial license agreement with Posit Software, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#ifndef TESTS_FIXTURES_REQUIRES_PRIVILEGE_TEST_FIXTURE_HPP
+#define TESTS_FIXTURES_REQUIRES_PRIVILEGE_TEST_FIXTURE_HPP
```

Comment:
This looks like it's being used by PosixSystemTest. It looks like it's more gracefully handling root constraint. Is that a real constraint anyway? Otherwise, the file looks fine.